bot/bot.go

```diff
@@ -171,6 +159,37 @@ Now, enjoy the community and have fun.`
 	return err
 }
 
+func (b *Bot) listChannels(ctx context.Context, span *trace.Span) ([]slack.Channel, error) {
+	childSpan := span.NewChild("slackApi.GetConversations")
+	childSpan.Finish()
+
+	params := &slack.GetConversationsParameters{
+		ExcludeArchived: "true",
+		Limit:           200,
+		Types: []string{
+			"public_channel",
+			"private_channl",
+		},
+	}
+
+	channels, nextCursor, err := b.slackBotAPI.GetConversationsContext(ctx, params)
```

Comment:
Suggestion:

```go
	var channels []slack.Channel
	for {
		c, nc, err := b.slackBotAPI.GetConversationsContext(ctx, params)
		if err != nil {
			return channels, err
		}
		if nc == "" {
			break
		}
		params.Cursor = nc
		channels = append(channels, c...)
	}
```

The [slack docs](https://api.slack.com/docs/pagination) specifically state that there are no more results when the next cursor is empty, this inlines that idea and eliminates the call to GetConversationsContext outside of the loop.